src/app/routes/storage.ts

```diff
@@ -292,6 +293,10 @@ export class StorageController {
 
         this.logger.error(`Error getting folder contents, folderId: ${id}: ${err}. Stack: ${err.stack}`);
 
+        if (err instanceof ConnectionTimedOutError) {
+          res.status(504).send();
```

Comment:
missing a return here, if the return is not placed, Node.js will throw a "Response headers have already been sent" error

Reply:
updated!